### PR TITLE
Enable mariadb for Basalt & rename it to mariadb

### PR DESCRIPTION
**Pull request by dcermak**
But keep rmt-mariadb on SLE for the helm chart compatibility

For [jsc#PED-2742](https://jira.suse.com/browse/PED-2742)

### CODE REVIEW

File: src/bci_build/package/appcontainers.py [hunk 1]
```diff
@@ -134,33 +134,45 @@
...
+            ],
+            pretty_name="MariaDB Server",
+            package_list=["mariadb", "mariadb-tools", "gawk", "timezone", "util-linux"],
+            entrypoint=["docker-entrypoint.sh"],
+            extra_files={
+                "docker-entrypoint.sh": _MARIAD_ENTRYPOINT,
+                "_constraints": generate_disk_size_constraints(11),
+            },
+            build_recipe_type=BuildType.DOCKER,
+            cmd=["mariadbd"],
```
Comment (dirkmueller) [inline src/bci_build/package/appcontainers.py:172]: this was `mariadb` before, why did we change to `mariadbd` ? is that the same?

**Review by dirkmueller — APPROVED** [review]

File: src/bci_build/package/appcontainers.py [hunk 2]
```diff
@@ -134,33 +134,45 @@
...
-        build_recipe_type=BuildType.DOCKER,
-        cmd=["mariadbd"],
-        volumes=["/var/lib/mysql"],
-        exposes_tcp=[3306],
-        custom_end=rf"""{DOCKERFILE_RUN} mkdir /docker-entrypoint-initdb.d
+MARIADB_CONTAINERS = []
+MARIADB_CLIENT_CONTAINERS = []
+
+for os_version in set(ALL_NONBASE_OS_VERSIONS) | {OsVersion.BASALT}:
+    if os_version == OsVersion.BASALT:
```
Comment (dirkmueller) [inline src/bci_build/package/appcontainers.py:141]: it should use the "new" name for tumbleweed also, not just for BASALT.

**Review by dirkmueller — CHANGES_REQUESTED** [review]

File: src/bci_build/package/appcontainers.py (hunk 2 above)
Comment (dcermak) [reply src/bci_build/package/appcontainers.py]: Makes sense, I have included Tumbleweed in here too.

**Review by dirkmueller — APPROVED** [review]
I'm not a fan of having a basalt mariadb and a sle15 mariadb container but its not the right place to discuss that here..

**Pull request merged**